### PR TITLE
add feature: custom field set

### PR DESCRIPTION
**Pull request by itwillwork**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,24 +6,40 @@ function format(val, figure) {
...
+    },
+    minute: function (date) {
+        return format(date.getMinutes(), 2);
+    },
+    second: function (date) {
+        return format(date.getSeconds(), 2);
+    },
+};
+
+function date2obj(dateObject, neededFields) {
```
Comment (tigercosmos) [inline index.js]: Could you change to the form:
``` js
function date2obj(dateObject [, options]) {
```
which `options` is:
```js
options = {
 fields: ["year", ...]
}
```
This enables new features in the future.

File: test.js [hunk 2]
```diff
@@ -2,9 +2,14 @@ const date2obj = require('./index');
...
 
 function testobj(data1, data2) {
+    assert.equal(Object.keys(data1).length, Object.keys(data2).length,
+        `strict not equal`);
+
     for (item in data1) {
         assert.equal(data1[item], data2[item],
             `${item} not equeal: ${data1[item]} and ${data2[item]}`);
+
+        return;
```
[the remark this answers is not in the document]
Comment (itwillwork) [reply test.js]: i agree

File: index.js (hunk 1 above)
Comment (itwillwork) [reply index.js]: i agree

**Comment by itwillwork** [comment]
all added)

**Comment by tracker1** [comment]
I'm not sure if this should be a feature... the overhead for the fields is pretty big, and doesn't add much value...

you can filter with Object.assign to undefined properties if you really need to

    Object.assign(date2obj(new Date()), {year:undefined, month:undefined, day:undefined, milliseconds:undefined})

File: .idea/workspace.xml [hunk 3]
```diff
@@ -0,0 +1,207 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tigercosmos) [inline .idea/workspace.xml]: This file should be removed. @itwillwork

**Comment by tigercosmos** [comment]
I think it is worth to be a feature, since we use module for convenience. Any enhancement is good.
The efficiency is what I am concern about..
anyway, we can launch feature first and refactor later.

File: .idea/workspace.xml (hunk 3 above)
Comment (tigercosmos) [reply .idea/workspace.xml]: @itwillwork are your here?

Comment (itwillwork) [reply .idea/workspace.xml]: @tigercosmos  removed

**Comment by tigercosmos** [comment]
@itwillwork  thanks

**Pull request merged**